en/lessons/specifics/debugging.md

```diff
@@ -8,6 +8,113 @@ In this lesson we'll learn about debugging Elixir code as well as static analysi
 
 {% include toc.html %}
 
+# IEx
+
+The most straightforward tool we have for debugging Elixir code is IEx. But don't be fooled by its simplicity - you can solve most of the issues with your application by it.
+
+IEx means `Elixir's interactive shell`.  You could have already seen IEx in one of the previous lessons like [Basics](../../basics/basics) where we ran Elixir code interactively in the shell.
+
+The idea here is simple. You get the interactive shell in the context of the place you want to debug.
+
+Let's try it. To do that, create a file named `test.exs` and put this into the file:
+
+```
+defmodule TestMod do
+  def sum([a, b]) do
+    b = 0
+
+    a + b
+  end
+end
+
+IO.puts(TestMod.sum([34, 65]))
+```
+
+And if you run it - you'll get an apparent output of `34`:
+
+```
+$ elixir test.exs
+warning: variable "b" is unused (if the variable is not meant to be used, prefix it with an underscore)
+  test.exs:2
+
+34
+```
+
+But now let's get to the exciting part - the debugging. Put `require IEx; IEx.pry` in the line after `b = 0` and let's try running it once again. You'll get something like this:
```

Comment:
Let's add a snippet here that shows the `sum/2` function with the `require IEx; IEx.pry` on the appropriate line.

Reply:
I don't think that's necessary. We would duplicate the same listing with one line changed. Why do you think this is unclear?

Reply:
Actually on second thought, you're right. I think it is clear enough as-is. Thanks!